pandas/core/indexes/multi.py

```diff
@@ -3415,7 +3420,7 @@ def union(self, other, sort=None):
         other, result_names = self._convert_can_do_setop(other)
 
         if len(other) == 0 or self.equals(other):
-            return self
+            return self._shallow_copy(names=result_names)
```

Comment:
self.rename?

Reply:
Yup, makes sense. Certainly more legible, and seems like equivalent.